src/main/java/de/picturesafe/search/samples/aggregations/SearchWithDateHistogramAggregation.java

```diff
@@ -0,0 +1,112 @@
+import de.picturesafe.search.elasticsearch.model.DocumentBuilder;
+import de.picturesafe.search.elasticsearch.model.ResultFacet;
+import de.picturesafe.search.elasticsearch.model.ResultFacetItem;
+import de.picturesafe.search.elasticsearch.model.SearchResult;
+import de.picturesafe.search.expression.Expression;
+import de.picturesafe.search.expression.FulltextExpression;
+import de.picturesafe.search.parameter.SearchParameter;
+import de.picturesafe.search.parameter.SortOption;
+import de.picturesafe.search.parameter.aggregation.DateHistogramAggregation;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.stereotype.Component;
+
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.Date;
+import java.util.stream.LongStream;
+
+import static de.picturesafe.search.parameter.aggregation.DateHistogramAggregation.IntervalType.CALENDAR;
+
+@Component
+@ComponentScan
+public class SearchWithDateHistogramAggregation {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SearchWithDateHistogramAggregation.class);
+
+    @Autowired
+    private SingleIndexElasticsearchService singleIndexElasticsearchService;
+
+    public static void main(String[] args) {
+        try (AnnotationConfigApplicationContext ctx = new AnnotationConfigApplicationContext(SearchWithDateHistogramAggregation.class)) {
+            final SearchWithDateHistogramAggregation facet = ctx.getBean(SearchWithDateHistogramAggregation.class);
+            facet.run();
+        }
+    }
+
+    private void run() {
+        try {
+            singleIndexElasticsearchService.createIndexWithAlias();
+            createTestRecords();
+
+            final Expression expression = new FulltextExpression("test title");
+
+            // Search results and retrieve first page
+            final SearchResult searchResult = singleIndexElasticsearchService.search(expression, createSearchParameter());
+            LOGGER.info("Found {} hits with {} facet(s)", searchResult.getResultCount(), searchResult.getFacets().size());
+            showFacets(searchResult);
+        } finally {
+            singleIndexElasticsearchService.deleteIndexWithAlias();
+        }
+    }
+
+    private void createTestRecords() {
+        final Instant now = Instant.now();
+        final Instant yesterday = now.minus(1, ChronoUnit.DAYS);
+        final Instant oneYearAgo = now.minus(365, ChronoUnit.DAYS);
+
+        // Insert 20 test records with create date 'now'
+        LongStream.range(1, 21)
+                .forEach(id -> singleIndexElasticsearchService.addToIndex(DataChangeProcessingMode.BLOCKING,
+                        DocumentBuilder.id(id).put("title", "This is a test title " + id).put("created", Date.from(now)).build()));
```

Comment:
A loop with single adds in blocking mode is a serious performance issue! I know this is no production code, but a sample should give a hint on how to use the API correctly.